public/app/features/datasources/components/picker/DataSourceList.tsx

```diff
@@ -32,88 +31,16 @@ export interface DataSourceListProps {
   onClear?: () => void;
 }
 
-/**
- * Component state description for the {@link DataSourceList}
- *
- * @internal
- */
-export interface DataSourceListState {
-  error?: string;
-}
-
-/**
- * Component to be able to select a datasource from the list of installed and enabled
- * datasources in the current Grafana instance.
- *
- * @internal
- */
-export class DataSourceList extends PureComponent<DataSourceListProps, DataSourceListState> {
-  dataSourceSrv = getDataSourceSrv();
-
-  static defaultProps: Partial<DataSourceListProps> = {
-    filter: () => true,
-  };
-
-  state: DataSourceListState = {};
-
-  constructor(props: DataSourceListProps) {
-    super(props);
-  }
-
-  componentDidMount() {
-    const { current } = this.props;
-    const dsSettings = this.dataSourceSrv.getInstanceSettings(current);
-    if (!dsSettings) {
-      this.setState({ error: 'Could not find data source ' + current });
-    }
-  }
-
-  onChange = (item: DataSourceInstanceSettings) => {
-    const dsSettings = this.dataSourceSrv.getInstanceSettings(item);
-
-    if (dsSettings) {
-      this.props.onChange(dsSettings);
-      this.setState({ error: undefined });
-    }
-  };
-
-  getDataSourceOptions() {
-    const { alerting, tracing, metrics, mixed, dashboard, variables, annotations, pluginId, type, filter, logs } =
-      this.props;
-
-    const options = this.dataSourceSrv.getList({
-      alerting,
-      tracing,
-      metrics,
-      logs,
-      dashboard,
-      mixed,
-      variables,
-      annotations,
-      pluginId,
-      filter,
-      type,
-    });
-
-    return options;
-  }
-
-  render() {
-    const { className, current } = this.props;
-    // QUESTION: Should we use data from the Redux store as admin DS view does?
-    const options = this.getDataSourceOptions();
-
-    return (
-      <div className={className}>
-        {options.map((ds) => (
-          <DataSourceCard
-            key={ds.uid}
-            ds={ds}
-            onClick={this.onChange.bind(this, ds)}
-            selected={!!isDataSourceMatch(ds, current)}
-          />
-        ))}
-      </div>
-    );
-  }
+export function DataSourceList(props: DataSourceListProps) {
+  const { className, current, onChange } = props;
+  // QUESTION: Should we use data from the Redux store as admin DS view does?
+  const getDataSources = useGetDatasources({ ...props });
+
+  return (
+    <div className={className}>
+      {getDataSources({ ...props }).map((ds) => (
```

Comment:
I'd be explicit here, and only pass the necessary props to the hook. 
```suggestion
  const { className, current, onChange, tracing, mixed } = props;
  // We could get the DS list directly here
  const datasources = useGetDatasources({ tracing, mixed });

  return (
    <div className={className}>
      {datasources.map((ds) => (
```